### PR TITLE
R volesti

### PR DESCRIPTION
**Pull request by TolisChal**
- We build the R package for CRAN submission

### CODE REVIEW

File: R-proj/DESCRIPTION [hunk 1]
```diff
@@ -1,19 +1,19 @@
 Package: volesti
 Type: Package
-License: GPL (>= 2)
-Title: Volume approximation of convex polytopes.
-Authors@R: c(person("Fisikopoulos", "Vissarion" , role=c("cph", "cre", "aut"), email="vissarion.fysikopoulos@oracle.com"),
+License: LGPL-3
+Title: Volume Approximation of Convex Polytopes
```
Comment (vissarion) [inline R-proj/DESCRIPTION]: add "Sampling" ?

File: R-proj/DESCRIPTION [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+Authors@R: c(person("Fisikopoulos", "Vissarion" , role=c("cph", "cre", "aut"), email="vissarion.fisikopoulos@gmail.com"),
              person("Chalkis", "Apostolos" , role=c("cph", "ctb", "aut"), email="tolis.chal@gmail.com"))
 Description: Package provides an R interface for VolEsti C++ package. VolEsti computes
              approximations of volume of polytopes given as a set of points or linear inequalities
              or Minkowski sum of segments (zonotopes). There are two algorithms for volume
              approximation as well as algorithms for sampling, rounding and rotating polytopes.
-Maintainer: Fisikopoulos Vissarion <vissarion.fysikopoulos@oracle.com>, Chalkis Apostolos <tolis.chal@gmail.com>
-Version: 1.0.0
+Maintainer: Fisikopoulos Vissarion <vissarion.fisikopoulos@gmail.com>, Chalkis Apostolos <tolis.chal@gmail.com>
+Version: 0.0.0
```
Comment (vissarion) [inline R-proj/DESCRIPTION]: version 1 wasn't ok?

File: R-proj/R/rand_rotate.R [hunk 3]
```diff
@@ -112,18 +90,41 @@ rand_rotate <- function(Inputs){
...
+  sam_arb_simplex = FALSE
+  sam_ball = FALSE
+  sam_sphere = FALSE
   #-------------------#
   
-  Mat = vol_R(A, W, e, Cheb_ball, annealing, win_len, N, C, ratio, frac, ball_walk, delta,
-              Vpoly, Zono, exact_zono, gen_only, Vpoly_gen, kind_gen, dim_gen, m_gen, round_only, 
-              rotate_only, sample_only, numpoints, variance, coordinate, rounding, verbose)
+  # set timer
+  tim = proc.time()
```
Comment (vissarion) [inline R-proj/R/rand_rotate.R]: in general timings are not needed in function declaration, the user can benchmark them if needed. So if you agree remove any timings from declarations. You can leave them in tests.

File: R-proj/man/CheBall.Rd [hunk 4]
```diff
@@ -2,7 +2,7 @@
 % Please edit documentation in R/CheBall.R
```
Comment (vissarion) [inline R-proj/man/CheBall.Rd]: should these files be part of the repo?

File: R-proj/man/vol_R.Rd [hunk 5]
```diff
@@ -0,0 +1,78 @@
+% Generated by hand
```
Comment (vissarion) [inline R-proj/man/vol_R.Rd]: why not from roxygen?

File: include/solve_lp.h [hunk 6]
```diff
@@ -25,7 +25,9 @@
 #include <stdio.h>
 #include <cmath>
 #include <exception>
-#include "run_headers/lp_lib.h"
+#undef Realloc
```
Comment (vissarion) [inline include/solve_lp.h]: why is this needed?

File: R-proj/DESCRIPTION (hunk 1 above)
Comment (TolisChal) [reply R-proj/DESCRIPTION]: done

File: R-proj/DESCRIPTION (hunk 2 above)
Comment (TolisChal) [reply R-proj/DESCRIPTION]: I think the first one has to be 0.0.0

File: R-proj/R/rand_rotate.R (hunk 3 above)
Comment (TolisChal) [reply R-proj/R/rand_rotate.R]: done

File: R-proj/man/CheBall.Rd (hunk 4 above)
Comment (TolisChal) [reply R-proj/man/CheBall.Rd]: Yes, they have to be included.

File: R-proj/man/vol_R.Rd (hunk 5 above)
Comment (TolisChal) [reply R-proj/man/vol_R.Rd]: Because RcppExports.R is auto generated in every build(). So roxygen comments are destroyed in every build().

File: include/solve_lp.h (hunk 6 above)
Comment (TolisChal) [reply include/solve_lp.h]: #undef Realloc
#undef Free

Are needed fro the conflicts between R.h and windows.h when we install the package at windows.

File: cran_gen/genCRANpkg.R [hunk 7]
```diff
@@ -0,0 +1,132 @@
+# create cran package folder
+dir.create("cran_package")
```
Comment (vissarion) [inline cran_gen/genCRANpkg.R]: I get this warning: ```Warning message:
In dir.create("cran_package") : 'cran_package' already exists```

File: R-proj/R/volume.R [hunk 8]
```diff
@@ -1,217 +1,196 @@
...
     }
   }
-  dimension = dim(A)[2] - 1
   
-  # [CV] initialization
-  win_len = 4 * ( dimension ^ 2 ) + 500
-  if (!is.null(Inputs$win_len)) {
-    win_len = Inputs$win_len
+  # set a too large vector for chebychev ball if it is not given as input
+  InnerB = rep(0, dimension + 5)
```
Comment (vissarion) [inline R-proj/R/volume.R]: not a good practice to put magic numbers (i.e. 5). why not computing inner ball here if missing and avoid tricks?

File: R-proj/src/Rvolume.cpp [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+        return HP;
+    }
+
+};
+
+
+// [[Rcpp::plugins(cpp11)]]
+// [[Rcpp::export]]
+double Rvolume (Rcpp::Reference P, Rcpp::NumericMatrix A, unsigned int walk_len, double e,
```
Comment (vissarion) [inline R-proj/src/Rvolume.cpp]: merge https://github.com/vissarion/volume_approximation/tree/R_volesti for a solution that reduce all those `if`'s to just one. Please do the same in all files (i.e. R-proj/src/Rsample_points.cpp etc)

File: R-proj/src/Rvolume.cpp [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+                Rcpp::NumericVector InnerBall, bool CG, unsigned int win_len,
+                unsigned int N, double C, double ratio, double frac,
+                bool ball_walk, double delta, bool Vpoly, bool Zono,
```
Comment (vissarion) [inline R-proj/src/Rvolume.cpp]: it is redundant to have that many `bool` parameters. One way to remove this is to add a field in the R reference object that represents the polytope to give info about its type (i.e. Hpolytope, Vpolytope etc) and use that in the if statements.

Comment (vissarion) [reply R-proj/src/Rvolume.cpp]: the above comment applies to all functions in  R-proj/src

File: R-proj/R/volume.R (hunk 8 above)
Comment (TolisChal) [reply R-proj/R/volume.R]: volume.R removed

File: R-proj/src/Rvolume.cpp (hunk 9 above)
Comment (TolisChal) [reply R-proj/src/Rvolume.cpp]: merge done.

File: R-proj/src/Rvolume.cpp (hunk 10 above)
Comment (TolisChal) [reply R-proj/src/Rvolume.cpp]: we now use int type
